src/System.Numerics.Matrices/src/System/Numerics/IMatrix.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace System.Numerics.Matrices
+{
+    interface IMatrix
+    {
+        double this[int col, int row] { get; set; }
```

Comment:
There are a few good ideas in the Rosyln project bouncing around. My favorite is the ability to specify operate compliance on a type. Additionally, for optimizations there would be way to enforce that a type was not only a struct but a blittlable struct for optimized `unsafe` work on it.

Example:
```
public struct Matrix2x2<T>
    where T: struct, unsafe,
    operator +(T, T) => T,
    operator -(T, T) => T,
    operator *(T, T) => T,
    operator /(T, T) => T
{
    // ...
}
```